datafusion/core/src/physical_plan/aggregates/mod.rs

```diff
@@ -360,7 +380,9 @@ impl ExecutionPlan for AggregateExec {
 
     fn required_input_distribution(&self) -> Vec<Distribution> {
         match &self.mode {
-            AggregateMode::Partial => vec![Distribution::UnspecifiedDistribution],
+            AggregateMode::Partial | AggregateMode::Single => {
```

Comment:
Shouldn't it be `Distribution::SinglePartition`?

Reply:
It should be the same as the `AggregateMode::Partial`.  And this new rule will be applied after the `EnforceDistribution` rule,  which means the distribution requirements are already satisfied, and then the adjacent `Partial` and `Final` Aggregate are combined to a new `Single` Aggregate.